lib/pipeline.js

```diff
@@ -147,6 +147,42 @@ class PipelineModel extends BaseModel {
     }
 
     /**
+     * Fetch all secrets that belong to this pipeline
+     * @property secrets
+     * @return Promise
+    */
+    get secrets() {
+        const listConfig = {
+            params: {
+                pipelineId: this.id
+            },
+            paginate: {
+                count: 25, // This limit is set by the matrix restriction
```

Comment:
Why would this be set by the matrix?  There can be 100+ secrets, unless you put that limit in the config-parser.